examples/list_slaves.rs

```diff
@@ -0,0 +1,35 @@
+//! EtherCAT master example: List slaves
+//!
+//! Opens master 0 (read-only), queries slave info for positions 0..31 and
+//! prints discovered slaves.
+use ethercat::{Master, SlavePos};
+use std::{io};
+
+pub fn main() -> Result<(), io::Error> {
+    // initialize logging
+    env_logger::init();
+
+    // open the first available master interface
+    let master = Master::open(0, ethercat::MasterAccess::ReadOnly)?;
+
+    log::info!("Master opened. Attempting to list slaves (positions 0..31)...");
+
+    let mut found = 0u32;
+    for i in 0u16..=31 {
+        let pos = SlavePos::from(i);
+        match master.get_slave_info(pos) {
+            Ok(info) => {
+                println!("Slave pos {}: {:?}", i, info);
+                found += 1;
+            }
+            Err(_) => {
```

Comment:
Ignoring errors quietly is not a good idea.

Reply:
Resolved in 56ffecab984ab83ed428a80aa8f8fa9d5b6d5da0